lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1224,6 +1237,7 @@ def exec_module(self, **kwargs):
                         network_profile=self.compute_models.NetworkProfile(
                             network_interfaces=nics
                         ),
+                        zones=zones
```

Comment:
In the SDK, `zones` requires [`str`], but you send [`int`]

https://github.com/Azure/azure-sdk-for-python/blob/a8b56b3b3a6f090cedb9e3e42921eef7d2cf7098/azure-mgmt-compute/azure/mgmt/compute/v2018_10_01/models/virtual_machine.py#L137 

Please also add an [integration test](https://github.com/ansible/ansible/blob/devel/test/integration/targets/azure_rm_virtualmachine/tasks/virtualmachine.yml)

Reply:
Good catch. Will update ASAP.

Reply:
Updated. Could use some help/hints on how to make the integrations tests I wrote work based on the current setup.